lib/BC/PcodeCFG.cpp

```diff
@@ -0,0 +1,166 @@
+#include <remill/BC/PCodeCFG.h>
+
+#include <algorithm>
+#include <cstddef>
+#include <sleigh/op.hh>
+#include <sleigh/opcodes.hh>
+#include <sleigh/pcoderaw.hh>
+#include <unordered_map>
+#include <variant>
+#include <vector>
+
+#include "lib/Arch/Sleigh/ControlFlowStructuring.h"
+
+namespace remill {
+namespace sleigh {
+
+PcodeCFG
+PcodeCFGBuilder::CreateCFG(const std::vector<RemillPcodeOp> &linear_ops) {
+  return PcodeCFGBuilder(linear_ops).Build();
+}
+
+
+std::vector<size_t> PcodeCFGBuilder::GetBlockStarts() {
+  std::vector<size_t> res;
+  res.push_back(0);
+
+
+  for (size_t curr_index = 0; curr_index < this->linear_ops.size();
+       curr_index += 1) {
+    auto interproc = this->GetIntraProcTargets(curr_index);
+    res.insert(res.end(), interproc.begin(), interproc.end());
+  }
+  return res;
+}
+
+PcodeBlock PcodeCFGBuilder::BuildBlock(size_t start_ind, size_t next_start) {
+  std::vector<RemillPcodeOp> ops;
+
+  for (size_t i = start_ind; i < next_start; i++) {
+    ops.push_back(this->linear_ops[i]);
+  }
+
+  return PcodeBlock(start_ind, std::move(ops),
+                    this->GetBlockExitsForIndex(next_start - 1));
+}
+
+
+namespace {
+struct IntraProcTransferCollector {
+  std::vector<size_t> operator()(const InstrExit &ex) {
```

Comment:
Cannot this class simply hold `std::vector< std::size_t >` as an attribute? Then there is no need to always return it (which needs to be later merged) but instead push can happen directly?

As an entry point a static method can be exposed that creates this class, runs the visit and then returns the attribute.

Reply:
I think it could probably could be a `void` operator yeah

Reply:
[b70c59d](https://github.com/lifting-bits/remill/pull/645/commits/b70c59da0de7aeca218c80dd979df05e0e33f21e) and [34f4113](https://github.com/lifting-bits/remill/pull/645/commits/34f41131c31fcaf8b5e3bcce22d7e2baacebdbef)